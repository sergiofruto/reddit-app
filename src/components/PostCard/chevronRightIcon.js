import React from 'react';

const ChevronRightIcon = () => {
  return (
    <svg viewBox="0 0 407.436 407.436" fill="white" width="16" height="16">
      <polygon points="112.814,0 91.566,21.178 273.512,203.718 91.566,386.258 112.814,407.436 315.869,203.718 " />
    </svg>
  );
};

export default ChevronRightIcon;