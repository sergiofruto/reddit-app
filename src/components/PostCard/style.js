import styled from 'styled-components';

export const Card = styled.div`
  padding: 10px 8px;
`;

export const CardTop = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
`;

export const Author = styled.h2`
  margin: 0;
  font-size: 16px;
  font-weight: 400;
  color: white;
`;

export const CardBody = styled.div`
  display: flex;
  align-items: center;
`;

export const CardBottom = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
`;