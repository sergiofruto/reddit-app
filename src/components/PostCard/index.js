import React, { useEffect, useState } from 'react';
import { useSelector, useDispatch } from "react-redux";
// import * as actions from '../actions/index';
import * as actions from '../../actions';

import { Card, CardTop, CardBody, CardBottom, Author} from './style';

const PostCard = ({ post }) => {
  const dispatch = useDispatch();

  // const [selectedPost, setSelectedPost] = useState([]);
  const handleSelectPost = (e, post) => {
    dispatch(actions.selectPost(post));
  };

  return (
    <Card onClick={(e) => handleSelectPost(e, post)}>
      <CardTop className="card-top">
        <Author>{post.author}</Author>
        <time>18 hours ago</time>
      </CardTop>
      <CardBody className="card-body">
        <img src={post.thumbnail} alt={post.title} />
        <h1>{post.title}</h1>
      </CardBody>
      <CardBottom className="card-bottom">
        <button>Dismiss post</button>
        <span>{post.num_comments} comments</span>
      </CardBottom>
    </Card>
  );
};

export default PostCard;