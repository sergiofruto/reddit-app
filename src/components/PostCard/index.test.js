import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import PostCard from '.';

let mockData = require('./../../__tests__/mock.json');
const mockStore = configureStore([]);

describe('PostCard React-Redux Component', () => {
  let store;
  let component;

  beforeEach(() => {
    store = mockStore({
      reddit: mockData.reddit,
    });
    component = renderer.create(
      <Provider store={store}>
        <PostCard post={mockData.reddit.posts[0].data}/>
      </Provider>
    );
  });

  it('should render with given state from Redux store', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should dispatch an action on button click', () => {

  });
});