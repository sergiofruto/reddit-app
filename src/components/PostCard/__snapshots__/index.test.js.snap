// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostCard React-Redux Component should render with given state from Redux store 1`] = `
<div
  className="sc-bdfBwQ jUntii"
  onClick={[Function]}
>
  <div
    className="sc-gsTCUz hoelXU"
  >
    <div
      className="sc-dlfnbm cQNAav"
    >
      <span
        className="sc-hKgILt jlJhHq"
      />
      <h2
        className="sc-eCssSg jXVPLx"
      >
        bentoboxkiller
      </h2>
      <span
        className="sc-jSgupP dKQLHt"
      >
        <time
          dateTime={"2021-01-21T20:26:24.000Z"}
        >
          10 hours ago
        </time>
      </span>
    </div>
    <div
      className="sc-gKsewC gpEKeI"
    >
      <img
        alt="Hey you finally awake"
        className="sc-iBPRYJ fjFCwn"
        src="https://b.thumbs.redditmedia.com/Pk5G9kE74kcDC9Ckkj65aCHqXyO7NFefq0aCXM8oKEw.jpg"
      />
      <h1
        className="sc-fubCfw gEgwfV"
      >
        Hey you finally awake
      </h1>
      <svg
        fill="white"
        height="16"
        viewBox="0 0 407.436 407.436"
        width="16"
      >
        <polygon
          points="112.814,0 91.566,21.178 273.512,203.718 91.566,386.258 112.814,407.436 315.869,203.718 "
        />
      </svg>
    </div>
    <div
      className="sc-pFZIQ kCSKrU"
    >
      <button
        className="sc-jrAGrp jySDZz"
        onClick={[Function]}
      >
        <svg
          fill="orange"
          height="20"
          space="preserve"
          viewBox="0 0 511.995 511.995"
          width="20"
        >
          <g>
            <g>
              <path
                d="M437.126,74.939c-99.826-99.826-262.307-99.826-362.133,0C26.637,123.314,0,187.617,0,256.005 s26.637,132.691,74.993,181.047c49.923,49.923,115.495,74.874,181.066,74.874s131.144-24.951,181.066-74.874 C536.951,337.226,536.951,174.784,437.126,74.939z M409.08,409.006c-84.375,84.375-221.667,84.375-306.042,0 c-40.858-40.858-63.37-95.204-63.37-153.001s22.512-112.143,63.37-153.021c84.375-84.375,221.667-84.355,306.042,0 C493.435,187.359,493.435,324.651,409.08,409.006z"
              />
            </g>
            <g>
              <path
                d="M341.525,310.827l-56.151-56.071l56.151-56.071c7.735-7.735,7.735-20.29,0.02-28.046 c-7.755-7.775-20.31-7.755-28.065-0.02l-56.19,56.111l-56.19-56.111c-7.755-7.735-20.31-7.755-28.065,0.02 c-7.735,7.755-7.735,20.31,0.02,28.046l56.151,56.071l-56.151,56.071c-7.755,7.735-7.755,20.29-0.02,28.046 c3.868,3.887,8.965,5.811,14.043,5.811s10.155-1.944,14.023-5.792l56.19-56.111l56.19,56.111 c3.868,3.868,8.945,5.792,14.023,5.792c5.078,0,10.175-1.944,14.043-5.811C349.28,331.117,349.28,318.562,341.525,310.827z"
              />
            </g>
          </g>
        </svg>
        Dismiss post
      </button>
      <span
        className="sc-kEjbxe fQooOP"
      >
        551
         comments
      </span>
    </div>
  </div>
</div>
`;
